src/pages/FiveCoinPuzzle.js

```diff
@@ -0,0 +1,201 @@
+import "../styles.css";
+import React, { useEffect, useState } from "react";
+import toast, { Toaster } from "react-hot-toast";
+import ArrowKeysReact from "arrow-keys-react";
+import Button from "../components/Button";
+import Coin from "../components/Coin";
+
+export default function FiveCoinPuzzle() {
+  const numberOfCoinSlots = 17;
+  const startingArray = [0, 0, 0, 0, 0, 0, 1, 2, 1, 2, 1, 0, 0, 0, 0, 0, 0];
+  // player wins if either of these segments are present at any point in the array
+  const winningArraySegments = [
+    [2, 2, 1, 1, 1],
+    [1, 1, 1, 2, 2],
+  ];
+  const [coinTypeArray, setCoinTypeArray] = useState(startingArray);
+  const [firstSelectedCoin, setFirstSelectedCoin] = useState(-1);
+  const [secondSelectedCoin, setSecondSelectedCoin] = useState(-1);
+  const [moveCounter, setMoveCounter] = useState(0);
+  const [coinSelectText, setcoinSelectText] = useState(
+    "Click to select first coin"
+  );
+
+  const swapCoins = (array, index1, index2) => {
+    [array[index1], array[index2]] = [array[index2], array[index1]];
+    setCoinTypeArray(array);
+    selectCoin(index2);
+    checkCompletion(array);
+  };
+  const selectCoin = (coinNumber) => {
+    // cannot select empty coins
+    if (coinTypeArray[coinNumber] !== 0) {
+      if (firstSelectedCoin === -1) {
+        setFirstSelectedCoin(coinNumber);
+        setcoinSelectText("Click to select second coin");
+      } else if (
+        secondSelectedCoin === -1 &&
+        (coinNumber === firstSelectedCoin - 1 ||
+          coinNumber === firstSelectedCoin + 1) &&
+        coinTypeArray[coinNumber] !== coinTypeArray[firstSelectedCoin]
```

Comment:
Pretty complicated-looking condition here. To clarify, it's enforcing the puzzle rules by checking that:
- we don't have a second coin selected already
- the coin being selected is adjacent to the first selected coin
- the coin being selected is the opposite color of the first selected coin

Any suggestions on making this more readable or is it okay?

Reply:
See above comment for a relatively simple way to make things more readable without reworking any fundamental logic :)